### PR TITLE
Add edge zone parameter  for vm image cmdlets

### PR DESCRIPTION
**Pull request by viananth**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/776
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/Images/GetAzureVMImageCommand.cs [hunk 1]
```diff
@@ -95,7 +102,69 @@ public override void ExecuteCmdlet()
 
             ExecuteClientAction(() =>
             {
-                if (this.ParameterSetName.Equals(ListVMImageParamSetName) || WildcardPattern.ContainsWildcardCharacters(Version))
+                if ((this.IsParameterBound(c => c.EdgeZone)) && this.EdgeZone != null)
+                {
+                    if (this.ParameterSetName.Equals(ListVMImageParamSetName) || WildcardPattern.ContainsWildcardCharacters(Version))
+                    {
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Images/GetAzureVMImageCommand.cs:108]: correct me if im wrong, but i think this if statement should be outside of if edge zone if statement, right?
What is customer wants to get ListofVMImage but without using edgezone?

Comment (grizzlytheodore) [reply src/Compute/Compute/Images/GetAzureVMImageCommand.cs:108]: never mind i see what you did at the bottom

File: src/Compute/Compute/Images/GetAzureVMImageCommand.cs [hunk 2]
```diff
@@ -95,7 +102,69 @@ public override void ExecuteCmdlet()
...
+                            this.Location.Canonicalize(),
+                            this.EdgeZone,
+                            this.PublisherName,
+                            this.Offer,
+                            this.Skus,
+                            version: this.Version).GetAwaiter().GetResult();
+
+                        var image = new PSVirtualMachineImageDetail
+                        {
+                            RequestId = response.RequestId,
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Images/GetAzureVMImageCommand.cs:147]: usually mapping of returned property from the HTTP call to PS object is done like this using mapper: https://github.com/Azure/azure-powershell/blob/master/src/Compute/Compute/Disk/NewAzDiskAccessCommand.cs#L57

I know this cmdlet was already like this, but I suggest using mapper instead going forward

**Review by grizzlytheodore — COMMENTED** [review]
looks good overall!!

**Comment by grizzlytheodore** [comment]
@viananth  please fill out the checklist up top, and also we will need to resolve the CI failure. 
you can follow the failed check links and press 'raw logs' at the end in the upper right corner, to get to this page: https://dev.azure.com/azure-sdk/29ec6040-b234-4e31-b139-33dc4287b756/_apis/build/builds/832550/logs/59

you can search for 'error' there and get some more information to troubleshoot

**Comment by grizzlytheodore** [comment]
after that if we are still unsure we can consult the powershell team

**Review by VeryEarly — CHANGES_REQUESTED** [review]
please update help markdown for these new parameters.

**Comment by viananth** [comment]
The build went fine, but the error is at save-artifacts:
`##[error]Artifact build-linux already exists for build 838414.`

**Comment by grizzlytheodore** [comment]
@MS-syh2qs @msJinLei  could you take a look at the CI failure? it seems strange to me, is it a pipeline issue?

**Comment by viananth** [comment]
@VeryEarly @grizzlytheodore Could you please merge if this looks good to you?

**Pull request merged**